packages/core/src/browser/frontend-application-config-provider.ts

```diff
@@ -33,14 +33,14 @@ export class FrontendApplicationConfigProvider {
             throw new Error('The configuration is already set.');
         }
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const globalObject = window as any;
+        const globalObject = globalThis as any;
```

Comment:
I was not aware of [`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis), nice. Unfortunately, it does not work as is:

```
 1) Preference Proxy
       "before all" hook for "by default, it should get provide access in flat style but not deep":
     ReferenceError: globalThis is not defined
      at Function.FrontendApplicationConfigProvider.doGet (lib/browser/frontend-application-config-provider.js:40:28)
      at Function.FrontendApplicationConfigProvider.set (lib/browser/frontend-application-config-provider.js:30:47)
      at Context.<anonymous> (lib/browser/preferences/preference-proxy.spec.js:86:82)

  2) Preference Service
       "before all" hook for "should return the preference from the more specific scope (user > workspace)":
     ReferenceError: globalThis is not defined
      at Function.FrontendApplicationConfigProvider.doGet (lib/browser/frontend-application-config-provider.js:40:28)
      at Function.FrontendApplicationConfigProvider.set (lib/browser/frontend-application-config-provider.js:30:47)
      at Context.<anonymous> (lib/browser/preferences/preference-service.spec.js:97:82)
```

Can we go back to `window` and change this in an another PR?

Reply:
Now I see why you did it:
```
App threw an error during load
ReferenceError: window is not defined
    at Function.FrontendApplicationConfigProvider.doGet (/Users/akos.kitta/git/theia/packages/core/lib/browser/frontend-application-config-provider.js:40:28)
    at Function.FrontendApplicationConfigProvider.set (/Users/akos.kitta/git/theia/packages/core/lib/browser/frontend-application-config-provider.js:30:47)
    at Object.<anonymous> (/Users/akos.kitta/git/theia/examples/electron/src-gen/frontend/electron-main.js:27:35)
    at Object.<anonymous> (/Users/akos.kitta/git/theia/examples/electron/src-gen/frontend/electron-main.js:68:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at loadApplicationPackage (/Users/akos.kitta/git/theia/node_modules/electron/dist/Electron.app/Contents/Resources/default_app.asar/main.js:121:12)

```

It wasn't an issue last time I have tried the changes, what's the difference?

Reply:
> what's the difference?

I've found it; we should not set the frontend app config form the electron main.

Reply:
OK, finally, I understood you why you've made these changes; let me propose something with a few fixes.